tools/link.py

```diff
@@ -574,25 +574,91 @@ def include_and_export(name):
       settings.EXPORTED_RUNTIME_METHODS += ['ExitStatus']
 
 
+def set_initial_memory():
+  user_specified_initial_heap = 'INITIAL_HEAP' in user_settings
+
+  # INITIAL_HEAP cannot be used when the memory object is created in JS.
+  if settings.IMPORTED_MEMORY:
+    if user_specified_initial_heap:
+      # Some of these could (and should) be implemented.
+      exit_with_error('INITIAL_HEAP is currently not compatible with IMPORTED_MEMORY, SHARED_MEMORY, RELOCATABLE, ASYNCIFY_LAZY_LOAD_CODE')
```

Comment:
How about `INITIAL_HEAP is currently not compatible with IMPORTED_MEMORY (which is enabled indirectly via  SHARED_MEMORY, RELOCATABLE, ASYNCIFY_LAZY_LOAD_CODE)`